src/core/scenario.jl

```diff
@@ -111,22 +111,77 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
+                                            time_steps_per_hour=settings.time_steps_per_hour
+                                        )
     elseif settings.off_grid_flag 
         if haskey(d, "ElectricUtility")
             @warn "ElectricUtility inputs are not applicable when off_grid_flag is true and any ElectricUtility inputs will be ignored. For off-grid scenarios, a year-long outage will always be modeled."
         end
-        electric_utility = ElectricUtility(; outage_start_time_step = 1, outage_end_time_step = settings.time_steps_per_hour * 8760) 
+        electric_utility = ElectricUtility(; outage_start_time_step = 1, 
+                                            outage_end_time_step = settings.time_steps_per_hour * 8760, 
+                                            latitude=site.latitude, longitude=site.longitude, 
+                                            time_steps_per_hour=settings.time_steps_per_hour
+                                        ) 
     end
 
+    #Handle missing emissions inputs (due to failed lookup and not provided by user)
+    #TODO: condition below code on: if site.off_grid == false
+    if isnothing(electric_utility.emissions_factor_series_lb_CO2_per_kwh)
+        if (!isnothing(site.CO2_emissions_reduction_min_pct) || 
+            !isnothing(site.CO2_emissions_reduction_max_pct) || 
+            settings.include_climate_in_objective)
+            throw(@error "To include CO2 costs in the objective function or enforce emissions reduction constraints, you must either enter custom health emissions factors or a site location within the continental U.S.")
+        else
+            electric_utility.emissions_factor_series_lb_CO2_per_kwh = zeros(8760*settings.time_steps_per_hour)
+        end
+    end
+    for emissions_type in ["NOx", "SO2", "PM25"]
+        if isnothing(getproperty(electric_utility, Symbol("emissions_factor_series_lb_$(emissions_type)_per_kwh")))
+            if settings.include_health_in_objective
+                throw(@error "To include health costs in the objective function, you must either enter custom health emissions factors or a site location within the continental U.S.")
+            else
+                setproperty!(electric_utility, 
+                    Symbol("emissions_factor_series_lb_$(emissions_type)_per_kwh"), 
+                    zeros(8760*settings.time_steps_per_hour)
+                )
+            end
+        end
+    end
+    #TODO: allow grid costs to be missing if site.off_grid == true
+    missing_health_inputs = false
+    for emissions_type in ["NOx", "SO2", "PM25"]
+        for health_input in [
+            "$(emissions_type)_grid_cost_per_tonne",
+            "$(emissions_type)_onsite_fuelburn_cost_per_tonne",
+            "$(emissions_type)_cost_escalation_pct"
+        ]
+            if ismissing(getproperty(financial, Symbol(health_input)))
+                missing_health_inputs = true
+                setproperty!(financial, Symbol(health_input), 0)
```

Comment:
why not default all of the `Financial` health costs to zero and avoid all this checking in the `Scenario` constructor? Would be nice to keep the `Financial` struct immutable for code integrity.

Reply:
see comment https://github.com/NREL/REopt.jl/pull/28/commits/926318dd0e86ca5b5522d9bc3ee6ad8e85e218f9#r903297193 for why not to make defaults zero